analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -283,6 +307,15 @@ private fun Graph.toPackageReferenceForest(root: Identifier): SortedSet<PackageR
 // See https://golang.org/ref/mod#pseudo-versions.
 private val PSEUDO_VERSION_REGEX = "^v0.0.0-(?:[\\d]{14}-(?<sha1>[0-9a-f]+)$)".toRegex()
 
+/** Separator string indicating that data of a new package follows in the output of the go mod why command. */
+private const val PACKAGE_SEPARATOR = "# "
+
+/**
```

Comment:
Why is chunking necesssary?  e.g. what problem does it avoid?

Reply:
I hoped to make this clear in the comment: to prevent an unrestricted, large output of a single command when there is a big number of dependencies. But I can try to make the comment more explicit.

Reply:
Would a very conservative upper bound be let's say 300 dependencies, with 300 lines per dependency, with 70 bytes per line, I'd arrive at a bit less than 7MB ? May this is really quite opinionated, so I'll leave it up to you. I don't have a strong opinion on that one.

pro:
- constant memory consumption
contra:
- a bit more complex
- a bit worse execution time

Reply:
I would argue that the `chunked()` function from the standard library makes it quite easy to handle these chunks, so the additional complexity is low. I always feel a bit uneasy if no size restrictions are applied. And there is #3860, which mentions analyzer results for go projects with a size of up to 10 GB.